clothes-detection/src/App.js

```diff
@@ -1,12 +1,121 @@
-import React from 'react';
+import React, {Component} from 'react';
 import './App.css';
+import MagicDropzone from 'react-magic-dropzone';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Hi, I am react app.</h1>
-    </div>
-  );
+function getImageAttributes(imageUri) {
+  const yurieApiKey = 'AIzaSyCc2ZRO2cPmW50mHI8SDG1okNRzDo33gAU';
+  const majorApiKey = 'AIzaSyB_LRUShuGlrxDwNvR9FBsyyiMUUGXZTb0';
+  const url = `https://vision.googleapis.com/v1/images:annotate?alt=json&key=${yurieApiKey}`;
+  const requestParams = {
+    "requests": [
+      {
+        "image": {
+          "content" : imageUri,
+          "source": {
+            "imageUri": ''
+          }
+        },
+        "features": [
+          {
+            "maxResults": 0,
+            "type": "LABEL_DETECTION"
+          }
+        ]
+      }
+    ]
+  }
+  const options = {
+    method: 'POST',
+    body: JSON.stringify(requestParams),
+  };
+  fetch(url, options)
+  .then(response => response.json())
+  .then(imageAttributes => {
+    // figure out how to pass imageAttributes out to the react components
+    let labelArr = imageAttributes.responses[0].labelAnnotations;
+    console.log(labelArr);
+  })
+  .catch(error => error.log(error))
+}
+
+
+// function renderImageResults() {
+
+// }
+class App extends Component {
+
+  // add data to manage inside this component (can mutate unlike props in functional component)
+  state = {
+    model: null,
+    preview: "",
+    predictions: []
+  };
+
+  // when model is always true (only for test)
+  componentDidMount() {
+    this.setState({
+      model: true
+    });
+  }
+
+  // assign URI of image when accepted ?
+    // console.log the value of state to understand the function of onDrop()
+  onDrop = (accepted, rejected, links) => {
+    this.setState({ 
+      preview: accepted[0].preview || links[0] },
+      function () {console.log(this.state.preview)
+    })
+  }
+
+  // convert local file => blob => base 64 (when uploaded image is local)
+  convertBase64 = (localFile) => {
+  fetch(localFile)
+  .then(function(response) {
+    return response.blob()
+  })
+  .then(function(blob) {
+    let reader = new FileReader(); 
+    reader.readAsDataURL(blob);
+    reader.onload = function () {
+      let rawBase64data = reader.result;
+      // remove the header string of base 64 data
+      let finalBase64data = rawBase64data.substr(rawBase64data.indexOf(',') + 1);
+      getImageAttributes(finalBase64data);
+    }
+  })
+  .catch(error => error.log(error))
+  }
+
```

Comment:
[**A comment** (PR comments with a purpose)](https://gist.github.com/echolabstech/a905ad02efba1ef6663c96ee4f251480):
create function `makeClotheDescription` which extracts labels
create function 'findClothesMatch` which looks into the datastore for a match using the labels. This might require writing an sql query, or a REST request with query params, etc...Depending on how you implement.

Reply:
Do you mean rewriting this code inside `makeClotheDescription function`? 
`let labelArr = imageAttributes.responses[0].labelAnnotations;`